### PR TITLE
Add reclaimspace annotation for rbd storageclass if encryption enabled

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
https://issues.redhat.com/browse/RHSTOR-5878

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
Screenshot from 2024-05-30 22-08-41

File: packages/odf/components/create-storage-system/footer.tsx [hunk 1]
```diff
@@ -325,7 +335,12 @@ const handleReviewAndCreateNext = async (
         shouldSetCephRBDAsDefault: shouldSetCephRBDAsDefault,
       });
 
-      await createStorageSystem(subSystemName, subSystemKind, systemNamespace);
+      await createStorageSystem(
+        subSystemName,
+        subSystemKind,
+        systemNamespace,
```
Comment (GowthamShanmugam) [inline packages/odf/components/create-storage-system/footer.tsx:341]: You are adding an annotation to the storage system, not a storage class. If  I am correct adding annotation to the default storage class has to be done from the OCS operator side.

**Comment by bipuladh** [comment]
make changes at https://github.com/red-hat-storage/odf-console/blob/master/plugins/odf/console-extensions.json#L353-L355
Update the mutator function

**Comment by TimothyAsirJeyasing** [comment]
image
image

File: packages/odf/components/create-storage-system/footer.tsx (hunk 1 above)
Comment (TimothyAsirJeyasing) [reply packages/odf/components/create-storage-system/footer.tsx]: Ack

File: packages/ocs/storage-class/mutators.ts [hunk 2]
```diff
@@ -34,3 +34,17 @@ export const addKubevirtAnnotations = (sc: StorageClass): StorageClass => {
...
+
+export const addReclaimSpaceAnnotation = (sc: StorageClass): StorageClass => {
+  sc.metadata.annotations = {
+    ...sc.metadata?.annotations,
+    'reclaimspace.csiaddons.openshift.io/schedule': '@weekly',
+  };
+  return sc;
+};
+
+export const applyAllMutators = (sc: StorageClass): StorageClass => {
```
Comment (GowthamShanmugam) [inline packages/ocs/storage-class/mutators.ts:46]:
```suggestion
export const addRBDAnnotations = (sc: StorageClass): StorageClass => {
```

**Comment by SanjalKatiyar** [comment]
add epic/story link to the description...

**Review by GowthamShanmugam — APPROVED** [review]

**Comment by SanjalKatiyar** [comment]
can u add YAML screenshot for encrypted StorageClass ??
also plz check why tests are failing...

**Comment by alfonsomthd** [comment]
LGTM

**Pull request merged**